app/src/main/java/com/vestrel00/daggerbutterknifemvp/ui/common/BaseFragment.java

```diff
@@ -53,13 +58,52 @@
     @Inject
     DispatchingAndroidInjector<Fragment> fragmentInjector;
 
+    @Nullable
+    private Unbinder viewUnbinder;
+
     @Override
     public void onAttach(Context context) {
         AndroidInjection.inject(this);
         super.onAttach(context);
     }
 
     @Override
+    public void onViewStateRestored(Bundle savedInstanceState) {
+        super.onViewStateRestored(savedInstanceState);
+        View view = getView();
+        if (view != null) {
+            /*
+             * Bind the views here instead of in onViewCreated so that view state changed listeners
+             * are not invoked automatically without user interaction.
+             *
+             * If we bind before this method (e.g. onViewCreated), then any checked changed
+             * listeners bound by ButterKnife will be invoked during fragment recreation (since
+             * Android itself saves and restores the views' states.
+             *
+             * The lifecycle order is as follows (same if added via xml or java or if retain
+             * instance is true):
+             *
+             * onAttach
+             * onCreateView
+             * onViewCreated
+             * onActivityCreated
+             * onViewStateRestored
+             * onResume
+             */
+            viewUnbinder = ButterKnife.bind(this, getView());
```

Comment:
Yes. I didn;t include `onStart` because it wasn't relevant here. I also omitted `onCreate` on purpose.